File: tests/integration/benchmarks/fixtures/go/rosetta/src/helpers/helpers_test.go
// Package helpers tests for the helpers module.
//
// This test file exercises name-matched test detection:
// helpers_test.go should automatically back helpers.go.
package helpers

import (
	"testing"
)

func TestFormat(t *testing.T) {
	t.Run("formats numbers with two decimal places", func(t *testing.T) {
		if Format(100) != "100.00" {
			t.Errorf("expected 100.00, got %s", Format(100))
		}
		if Format(0) != "0.00" {
			t.Errorf("expected 0.00, got %s", Format(0))
		}
	})
}

func TestSum(t *testing.T) {
	t.Run("sums a slice of numbers", func(t *testing.T) {
		result := Sum([]float64{1, 2, 3, 4, 5})
		if result != 15 {
			t.Errorf("expected 15, got %f", result)
		}

		result = Sum([]float64{100, 200, 300})
		if result != 600 {
			t.Errorf("expected 600, got %f", result)
		}
	})

	t.Run("returns 0 for empty slice", func(t *testing.T) {
		result := Sum([]float64{})
		if result != 0 {
			t.Errorf("expected 0, got %f", result)
		}
	})
}

func TestAverage(t *testing.T) {
	t.Run("calculates average of numbers", func(t *testing.T) {
		result := Average([]float64{10, 20, 30})
		if result != 20 {
			t.Errorf("expected 20, got %f", result)
		}
	})

	t.Run("returns 0 for empty slice", func(t *testing.T) {
		result := Average([]float64{})
		if result != 0 {
			t.Errorf("expected 0, got %f", result)
		}
	})
}

File: tests/integration/benchmarks/fixtures/java/rosetta/src/com/rosetta/helpers/HelpersTest.java
package com.rosetta.helpers;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Nested;
import org.junit.jupiter.api.Test;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

import static org.junit.jupiter.api.Assertions.*;

/**
 * Unit tests for the Helpers class.
 *
 * This test file exercises name-matched test detection:
 * HelpersTest.java should automatically back Helpers.java.
 */
public class HelpersTest {

    @Nested
    @DisplayName("format()")
    class FormatTests {

        @Test
        @DisplayName("formats numbers with two decimal places")
        void formatsNumbersWithTwoDecimalPlaces() {
            assertEquals("100.00", Helpers.format(100.0));
            assertEquals("0.00", Helpers.format(0.0));
            assertEquals("123.46", Helpers.format(123.456));
        }
    }

    @Nested
    @DisplayName("sum()")
    class SumTests {

        @Test
        @DisplayName("sums a list of numbers")
        void sumsListOfNumbers() {
            List<Double> values = Arrays.asList(1.0, 2.0, 3.0, 4.0, 5.0);
            assertEquals(15.0, Helpers.sum(values), 0.001);
        }

        @Test
        @DisplayName("returns 0 for empty array")
        void returnsZeroForEmptyArray() {
            assertEquals(0.0, Helpers.sum(Collections.emptyList()), 0.001);
        }
    }

    @Nested
    @DisplayName("average()")
    class AverageTests {

        @Test
        @DisplayName("calculates average of numbers")
        void calculatesAverageOfNumbers() {
            List<Double> values = Arrays.asList(10.0, 20.0, 30.0);
            assertEquals(20.0, Helpers.average(values), 0.001);
        }

        @Test
        @DisplayName("returns 0 for empty array")
        void returnsZeroForEmptyArray() {
            assertEquals(0.0, Helpers.average(Collections.emptyList()), 0.001);
        }
    }
}
